### PR TITLE
chore: add the possibility to disable logging and tracing

### PR DESCRIPTION
**Pull request by marinelli**
This PR proposes to add settings to disable logging and/or tracing when starting a client.

### CODE REVIEW

File: src/Otel/Internal/Client.hs [hunk 1]
```diff
@@ -82,26 +86,25 @@ data OtelClient = OtelClient
...
+  traceRequestBase <- setProtobufRequest <$> parseRequest traceEndpoint
   let resourceAttributes = toOtelAttributes resourceAttributes'
   logQueueSet <- newOtelQueueSetIO queueSize
   traceQueueSet <- newOtelQueueSetIO queueSize
   let client = OtelClient {..}
-  void . forkIO $ runLogClientProcess client
-  void . forkIO $ runTraceClientProcess client
+  when (not logEnabled && not traceEnabled) $
+    putStrLn "WARNING: Logging and Tracing are disabled."
+  when logEnabled . void . forkIO $ runLogClientProcess client
```
Comment (Siprj) [inline src/Otel/Internal/Client.hs:99]: If I recall correctly, this is not enough. You need to disable the insertion into the queues as well, or else the app will hang when the queues will fill.

Comment (marinelli) [reply src/Otel/Internal/Client.hs:99]: I forgot to patch `runOtelStatic` and the `log` function. I have to admit that I am not sure the current code is good enough. I'm open to hearing proposals :)

**Comment by Siprj** [comment]
Looks good. Thanks!

**Comment by marinelli** [comment]
> Looks good. Thanks!

Thanks for taking a look.

**Pull request merged**